### PR TITLE
Owls 99380 - Fix for pod roll issue when domain upgraded from 3.4 to 4.0 and conversion webhook pre-created

### PR DESCRIPTION
**Pull request by ankedia**
OWLS-99380 - Fix for pod roll issue when domain is upgraded from 3.4 to 4.0 and conversion webhook is pre-created using `webhookOnly` option.

This change performs the additional/regular hash comparison for the legacy aux images when the hash comparation after adjusting the names of the compatibility-mode init-container and the volume/mount fails. I will be creating a second PR for the 3.4 branch that will have changes to create legacy aux image volume/mounts and environment variables in a specific order. 
This PR also contains changes to the integration tests to temporarily use the operator 3.4 image (containing the release/3.4 changes) from OCIR instead of from ghcr.io. This change avoids the need for upgrade tests to `git clone` two branches and the change will need to be reverted after next 3.4.x release is available.

Integration test results - 
Upgrade test results - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10647 
Complete suite - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10673/
3.4 release test results - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10674/

### CODE REVIEW

**Review by rjeberhard — APPROVED** [review]

**Review by anpanigr — COMMENTED** [review]
Should we not support this upgrade path from release/3.3.x to 4.0
I see only ItOperatorWlsUpgrade#testOperatorWlsAuxDomainUpgradeFrom340ToCurrent test in the jenkin run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10647

**Comment by ankedia** [comment]
> Should we not support this upgrade path from release/3.3.x to 4.0 I see only ItOperatorWlsUpgrade#testOperatorWlsAuxDomainUpgradeFrom340ToCurrent test in the jenkin run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10647
[ANIL] We only support the interop/coexistence of 3.4 and 4.0 and we do not support the coexistence of 3.3.x and 4.0. For the upgrade path from release/3.3.x to 4.0, the conversion webhook should not be pre-created.

**Comment by anpanigr** [comment]
> > Should we not support this upgrade path from release/3.3.x to 4.0 I see only ItOperatorWlsUpgrade#testOperatorWlsAuxDomainUpgradeFrom340ToCurrent test in the jenkin run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10647
> > [ANIL] We only support the interop/coexistence of 3.4 and 4.0 and we do not support the coexistence of 3.3.x and 4.0. For the upgrade path from release/3.3.x to 4.0, the conversion webhook should not be pre-created.

Please remove the test testOperatorWlsAuxDomainUpgradeFrom388ToCurrent if we do not support this usecase.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/OperatorUtils.java [hunk 1]
```diff
@@ -47,6 +47,10 @@
 
 public class OperatorUtils {
 
+  public static final String OPERATOR_VERSION_340 = "3.4.0";
+  public static final String OPERATOR_340_IMAGE_OWLS_99380 =
+      "phx.ocir.io/weblogick8s/weblogic-kubernetes-operator:owls_99380";
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/OperatorUtils.java:52]: Is this a public image? Is it accessible to anyone without authentication?

Comment (ankedia) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/OperatorUtils.java:52]: Yes.

**Comment by ankedia** [comment]
> Please remove the test testOperatorWlsAuxDomainUpgradeFrom388ToCurrent if we do not support this usecase.
Disabled in [dd4178e](https://github.com/oracle/weblogic-kubernetes-operator/pull/3085/commits/dd4178edd0c3ecc5670ad3510672647eaf013470).

**Review by anpanigr — APPROVED** [review]
lgtm

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**